File: 876_leet/ans.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
 struct ListNode {
    int val;
     struct ListNode *next;
 };

int lenghtOfList(struct ListNode* head)
{
    struct ListNode* temp = head;
    int count = 0;

    while (temp)
    {
        count += 1;
        temp = temp->next;
    }
    return (count);
}

struct ListNode* middleNode(struct ListNode* head){
    int lenght = 0, middle = 0;
    struct ListNode* temp = head;

    lenght = lenghtOfList(head);
    middle = lenght / 2;

    while(middle > 0)
    {
        temp = temp->next;
        middle -= 1;
    }
    
    return (temp);
}

File: 876_leet/ans.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var middleNode = function(head) {
    var lenght = 0;
    var middle = 0;
    var temp = head;

    while (temp)
    {
        lenght++;
        temp = temp.next;
    }

    middle = Math.floor(lenght / 2);
    temp = head;

    while (middle)
    {
        middle--;
        temp = temp.next;
    }
    
    return temp;
};